src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -97,29 +95,27 @@ public TopicPartitionChannel(
 
     this.streamingBuffer = new StreamingBuffer();
     this.processedOffset = new AtomicLong(-1);
-    this.committedOffset = new AtomicLong(0);
+    this.offsetSafeToCommitToKafka = new AtomicLong(0);
```

Comment:
should this also start with an invalid value (like -1)?

Reply:
No, actually since offsets start from 0 in kafka, by default is 0. if we send -1, it will throw up!